js/common/lib/tensor.ts

```diff
@@ -234,5 +234,71 @@ export interface TensorConstructor {
   // #endregion
 }
 
+export interface ImageFromTensorOptions {
+  format?: 'RGB'|'RGBA'|'BGR'|'RBG';
+  tensorLayout?: 'NHWC'|'NCHW';
+  height?: number;
+  width?: number;
+  norm?: {
+    normBias?: number;  // Todo add support - |[number,number,number]|[number,number,number,number];
+    normMean?: number;
+  };  // Todo add support - |[number,number,number]|[number,number,number,number];
+}
+export interface TensorFromImageOptions {
+  bitmapFormat?: 'RGB'|'RGBA'|'BGR'|'RBG';  // used only in the case of image bitmap is used for
+  tensorFormat?: 'RGB'|'RGBA'|'BGR'|'RBG';
+  dataType?: 'float32'|'uint8';
+  tensorLayout?: 'NHWC'|'NCHW';
+  height?: number;
+  width?: number;
+  tensorHeight?: number;
+  tensorWidth?: number;
+  norm?: {
+    normBias?: number;  // Todo add support - |[number,number,number]|[number,number,number,number];
+    normMean?: number;
+  };  // Todo add support - |[number,number,number]|[number,number,number,number];
+}
+export interface TensorFactory {
+  /**
+   * create a tensor from image object - HTMLImageElement, ImageData, ImageBitmap, URL
+   *
+   * @param image - ImageData Type - composeed of: Uint8ClampedArray, width. height - uses known pixel format RGBA
+   * @param Options - Optional - Interface describing input image & output tensor -
+   * Input Defaults: RGBA, 3 channels, 0-255, NHWC - Output Defaults: same as input parameters
+   * @returns A promise that resolves to a tensor object
+   */
+  fromImage(image: ImageData, Options?: TensorFromImageOptions): Promise<Tensor>;
+
+  /**
+   * create a tensor from image object - HTMLImageElement, ImageData, ImageBitmap, URL
+   *
+   * @param image - HTMLImageElement Type - since the data is stored as ImageData no need for format parameter
+   * @param Options - Optional - Interface describing input image & output tensor -
+   * Input Defaults: RGBA, 3 channels, 0-255, NHWC - Output Defaults: same as input parameters
+   * @returns A promise that resolves to a tensor object
+   */
+  fromImage(image: HTMLImageElement, Options?: TensorFromImageOptions): Promise<Tensor>;
```

Comment:
use a more detailed param name.

`image` -> `imageElement` in this context.

Same to `url`, `imageBitmap` below

Reply:
use camelCase for parameter.

`Options` -> `options`

should apply to other overrides.